File: app/src/main/java/com/kota/Telnet/TelnetCommand.java
package com.kota.Telnet;

import androidx.annotation.NonNull;

public class TelnetCommand {
    public static final byte AUTH = 37;
    public static final byte DO = -3;
    public static final byte DONT = -2;
    public static final byte ECHO = 1;
    public static final byte IAC = -1;
    public static final byte IS = 0;
    public static final byte NAWS = 31;
    public static final byte NEW_ENV = 39;
    public static final byte SB = -6;
    public static final byte SE = -16;
    public static final byte SG = 3;
    public static final byte TERMINAL_TYPE = 24;
    public static final byte WILL = -5;
    public static final byte WONT = -4;
    public byte action = 0;
    public byte header = 0;
    public byte option = 0;

    public TelnetCommand() {
    }

    public TelnetCommand(byte aHeader, byte aAction, byte aOption) {
        this.header = aHeader;
        this.action = aAction;
        this.option = aOption;
    }

    public boolean isEqualTo(TelnetCommand aCommand) {
        return aCommand.header == this.header && aCommand.action == this.action && aCommand.option == this.option;
    }

    public boolean isEqualTo(int aHeader, int aAction, int aOption) {
        return aHeader == this.header && aAction == this.action && aOption == this.option;
    }

    public String getCommandNameString(int aCommand) {
        switch (aCommand) {
            case -16:
                return "SE";
            case -6:
                return "SB";
            case -5:
                return "WILL";
            case -4:
                return "WONT";
            case -3:
                return "DO";
            case -2:
                return "DONT";
            case -1:
                return "IAC";
            case 0:
                return "IS";
            case 1:
                return "ECHO";
            case 3:
                return "SG";
            case 24:
                return "TERMINAL_TYPE";
            case 31:
                return "NAWS";
            case 37:
                return "AUTH";
            case 39:
                return "NEW_ENV";
            default:
                return "UNKNOW(" + aCommand + ")";
        }
    }

    @NonNull
    public String toString() {
        return getCommandNameString(this.header) + "," + getCommandNameString(this.action) + "," + getCommandNameString(this.option);
    }
}

File: app/src/main/java/com/kota/telnet/TelnetCommand.kt
package com.kota.telnet

/* loaded from: classes.dex */
class TelnetCommand {
    var action: Byte
    var header: Byte
    var option: Byte

    constructor() {
        this.header = 0.toByte()
        this.action = 0.toByte()
        this.option = 0.toByte()
    }

    constructor(aHeader: Byte, aAction: Byte, aOption: Byte) {
        this.header = 0.toByte()
        this.action = 0.toByte()
        this.option = 0.toByte()
        this.header = aHeader
        this.action = aAction
        this.option = aOption
    }

    fun isEqualTo(aCommand: TelnetCommand): Boolean {
        return aCommand.header == this.header && aCommand.action == this.action && aCommand.option == this.option
    }

    fun isEqualTo(aHeader: Int, aAction: Int, aOption: Int): Boolean {
        return aHeader == this.header.toInt() && aAction == this.action.toInt() && aOption == this.option.toInt()
    }

    fun getCommandNameString(aCommand: Int): String {
        when (aCommand) {
            -16 -> return "SE"
            -6 -> return "SB"
            -5 -> return "WILL"
            -4 -> return "WONT"
            -3 -> return "DO"
            -2 -> return "DONT"
            -1 -> return "IAC"
            0 -> return "IS"
            1 -> return "ECHO"
            3 -> return "SG"
            24 -> return "TERMINAL_TYPE"
            31 -> return "NAWS"
            37 -> return "AUTH"
            39 -> return "NEW_ENV"
            else -> {
                val name = "UNKNOW(" + aCommand + ")"
                return name
            }
        }
    }

    override fun toString(): String {
        return getCommandNameString(this.header.toInt()) + "," + getCommandNameString(this.action.toInt()) + "," + getCommandNameString(
            this.option.toInt()
        )
    }

    companion object {
        const val AUTH: Byte = 37
        val DO: Byte = -3
        val DONT: Byte = -2
        const val ECHO: Byte = 1
        val IAC: Byte = -1
        const val IS: Byte = 0
        const val NAWS: Byte = 31
        const val NEW_ENV: Byte = 39
        val SB: Byte = -6
        val SE: Byte = -16
        const val SG: Byte = 3
        const val TERMINAL_TYPE: Byte = 24
        val WILL: Byte = -5
        val WONT: Byte = -4
    }
}
